### PR TITLE
Add automatic prebuilds

### PR DESCRIPTION
**Pull request by juliangruber**
In #17 and #18 I set up CI infrastructure on Linux, OSX and Windows, now with the addition of the `upload` script in this PR, whenever a commit is detected as changing the pkg's `.version` - read, publishing a new version - it will upload the prebuild for the current version to GitHub.

The only further steps necessary from @maxogden are that you set `PREBUILD_TOKEN` as env vars in both your travis and your appveyor account.

If we agree this approach is sound and works, I'll package it into a library for generic use across native modules.

As a small taste, this has been uploaded automatically:

screen shot 2016-11-27 at 16 34 10

### CODE REVIEW

File: upload.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+function prebuild (args, cb) {
+  var cli = join(__dirname, 'node_modules', '.bin', 'prebuild')
+  var ps = spawn(cli, args)
+  ps.stdout.pipe(process.stdout, { end: false })
+  ps.stderr.pipe(process.stderr, { end: false })
+  ps.on('exit', cb)
+}
+
+if (version('HEAD') === version('HEAD~1')) process.exit(0)
```
Comment (ralphtheninja) [inline upload.js]: Aha, bail out if version in package.json hasn't changed. Nice one :)

Comment (ralphtheninja) [inline upload.js]: Maybe?

```js
prebuild([
  '-b', process.version,
  '-u', process.env.PREBUILD_TOKEN
], process.exit)
```

File: upload.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+var execSync = require('child_process').execSync
+var spawn = require('win-spawn')
```
Comment (ralphtheninja) [inline upload.js]: `win-spawn` seems to be deprecated

File: upload.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+if (version('HEAD') === version('HEAD~1')) process.exit(0)
+
+prebuild([
+  '-b', process.version,
+  '-u', process.env.PREBUILD_TOKEN
+], function (code) {
+  process.exit(code)
+})
```
Comment (ralphtheninja) [inline upload.js]: Maybe?

```js
prebuild([
  '-b', process.version,
  '-u', process.env.PREBUILD_TOKEN
], process.exit)
```

File: upload.js (hunk 2 above)
Comment (juliangruber) [reply upload.js]: good catch! seems they didn't publish the readme with the deprecation warning. will replace with https://github.com/IndigoUnited/node-cross-spawn

**Comment by juliangruber** [comment]
also looks like windows is missing x64 builds

File: appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+install:
+  - ps: Install-Product node $env:nodejs_version
+  - npm install
+
+test_script:
+  - node --version
+  - npm --version
+  - npm test
+  - npm run upload
```
Comment (vweevers) [inline appveyor.yml]: What happens if someone does a PR with a change to `package.json`? Secure variables (i.e. the GH token) are not decoded for public PRs. Maybe add:

```yaml
branches:
  only:
    - master
```

Or an equivalent in js.

Might also wanna add [`skip_tags: true`](https://www.appveyor.com/docs/branches/#build-on-tags-github-and-gitlab-only).

**Comment by vweevers** [comment]
> also looks like windows is missing x64 builds

@juliangruber in my experience windows prebuilds required an explicit `--arch` flag.

File: upload.js (hunk 3 above)
Comment (juliangruber) [reply upload.js]: thought about it too, but this is more explicit and the file is not that big anyway

File: appveyor.yml (hunk 4 above)
Comment (juliangruber) [reply appveyor.yml]: You mean, the env var isn't set when someone else pushes?

**Pull request merged**